### PR TITLE
Fix incorrect snowflake serialization

### PR DESCRIPTION
**Pull request by MCausc78**
# Description

<title>

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue) 

# Checklist:

- [ ] Ran `dart analyze` or `make analyze` and fixed all issues
- [ ] Ran `dart format --set-exit-if-changed -l 160 ./lib` or `make format` and fixed all issues
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: lib/src/builders/message/component.dart [hunk 1]
```diff
@@ -98,7 +98,7 @@ class ButtonBuilder extends MessageComponentBuilder {
         if (label != null) 'label': label,
         if (emoji != null)
           'emoji': {
-            'id': emoji!.id == Snowflake.zero ? null : emoji!.id,
+            'id': emoji!.id == Snowflake.zero ? null : emoji!.id.toString(),
```
Comment (Rapougnac) [inline lib/src/builders/message/component.dart:101]:
```suggestion
            'id': emoji!.id == Snowflake.zero ? null : emoji!.id.value.toString()
```

File: lib/src/builders/message/component.dart [hunk 2]
```diff
@@ -223,7 +223,7 @@ class SelectMenuOptionBuilder extends CreateBuilder<SelectMenuOption> {
         if (description != null) 'description': description,
         if (emoji != null)
           'emoji': {
-            'id': emoji!.id.value,
+            'id': emoji!.id.toString(),
```
Comment (Rapougnac) [inline lib/src/builders/message/component.dart:226]:
```suggestion
            'id': emoji!.id.value.toString(),
```

**Review by Rapougnac — CHANGES_REQUESTED** [review]
This aint gonna work with the current implementation, as Snowflake mix-in ToStringHelper

**Review by Rapougnac — APPROVED** [review]
Ignore what I've said, I'm def stupid

**Review by abitofevrything — APPROVED** [review]

**Pull request merged**